packages/shared/src/components/fields/MarkdownInput/index.tsx

```diff
@@ -170,6 +170,119 @@ function MarkdownInput(
+      } as React.FormEvent<HTMLTextAreaElement>);
+
+      requestAnimationFrame(() => {
+        textarea.value = newValue;
+        textarea.setSelectionRange(newStart, newEnd);
+      });
+    };
+
+    const handleTextFormatting = (symbol: string) => {
+      const selectedValue = value.substring(selectionStart, selectionEnd);
+      const trimmedValue = selectedValue.trim();
+
+      const leadingWhitespace =
+        selectedValue.length - selectedValue.trimStart().length;
+      const newStart = selectionStart + leadingWhitespace;
+      const newEnd = newStart + trimmedValue.length;
+      const symbolLength = symbol.length;
+
+      const isFormatted =
+        value.substring(newStart - symbolLength, newStart) === symbol &&
+        value.substring(newEnd, newEnd + symbolLength) === symbol;
+
+      let updatedValue;
+      let updatedStart;
+      let updatedEnd;
+
+      if (isFormatted) {
+        updatedValue =
+          value.substring(0, newStart - symbolLength) +
+          trimmedValue +
+          value.substring(newEnd + symbolLength);
+        updatedStart = newStart - symbolLength;
+        updatedEnd = newEnd - symbolLength;
+      } else {
+        updatedValue =
+          value.substring(0, newStart) +
+          symbol +
+          trimmedValue +
+          symbol +
+          value.substring(newEnd);
+        updatedStart = newStart + symbolLength;
+        updatedEnd = newEnd + symbolLength;
+      }
+
+      updateTextarea(updatedValue, updatedStart, updatedEnd);
+    };
+
+    const handleLinkPaste = () => {
+      navigator.clipboard.readText().then((clipboardText) => {
+        const selectedText = value.substring(selectionStart, selectionEnd);
+        const trimmedText = selectedText.trim();
+
+        const leadingWhitespace =
+          selectedText.length - selectedText.trimStart().length;
+        const trailingWhitespace =
+          selectedText.length - selectedText.trimEnd().length;
+
+        const newStart = selectionStart + leadingWhitespace;
+        const newEnd = selectionEnd - trailingWhitespace;
+
+        const newText =
+          /^https?:\/\//.test(clipboardText) && trimmedText
+            ? `${value.substring(
+                0,
+                newStart,
+              )}[${trimmedText}](${clipboardText})${value.substring(newEnd)}`
+            : value.substring(0, selectionStart) +
+              clipboardText +
+              value.substring(selectionEnd);
+
+        const linkEnd = newStart + `[${trimmedText}](${clipboardText})`.length;
+        updateTextarea(newText, linkEnd, linkEnd);
+      });
+    };
+
+    if (e.metaKey || e.ctrlKey) {
+      switch (e.key) {
+        case 'b':
+          e.preventDefault();
+          handleTextFormatting('**');
+          break;
+        case 'i':
+          e.preventDefault();
+          handleTextFormatting('_');
+          break;
+        case 'v':
+          e.preventDefault();
+          handleLinkPaste();
```

Comment:
Would it not work if you execute `onLinkCommand` instead?

Reply:
@sshanzel Thanks for your review 🙏. I have used `onLinkCommand` when the user presses <kbd>ctrl+l</kbd> after they have selected the word and wants to insert link manually. In this case executing simply `onLinkCommand` would work, pressing <kbd>ctrl+l</kbd> will do this `[word](url)` . But for <kbd>ctrl + v</kbd> it's different if the user have selected the word and have pasted a `url` above it than we want to do this `[word](https://www.lipsum.com/)` otherwise if it's not a `url` than it's a normal paste. so that's why i have extracted the logic in `handleLinkPaste();` function. hope this makes sense 😊

Reply:
Right. Good point. Thank you for elaborating the point! 🚀